clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/cache/CacheRepository.java

```diff
@@ -0,0 +1,213 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.cloudfoundry.cache;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.fasterxml.jackson.databind.introspect.AnnotatedClass;
+import com.fasterxml.jackson.databind.introspect.JacksonAnnotationIntrospector;
+import com.netflix.spinnaker.cats.cache.Cache;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.cats.cache.RelationshipCacheFilter;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.model.*;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.netflix.spinnaker.clouddriver.cloudfoundry.cache.Keys.Namespace.*;
+import static java.util.Collections.emptySet;
+import static java.util.stream.Collectors.toSet;
+
+@Repository
+public class CacheRepository {
+  private final ObjectMapper objectMapper = new ObjectMapper()
+    .disable(MapperFeature.DEFAULT_VIEW_INCLUSION);
+
+  private final Cache cacheView;
+
+  public CacheRepository(Cache cacheView) {
+    this.cacheView = cacheView;
+    this.objectMapper
+      .setConfig(objectMapper.getSerializationConfig().withView(Views.Cache.class))
+      .setAnnotationIntrospector(new JacksonAnnotationIntrospector() {
+        @Override
+        public JsonPOJOBuilder.Value findPOJOBuilderConfig(AnnotatedClass ac) {
+          return new JsonPOJOBuilder.Value("build", "");
+        }
+      });
+  }
+
+  public Set<CloudFoundryApplication> findApplicationsByKeys(Collection<String> keys, Detail detail) {
+    return cacheView.getAll(APPLICATIONS.getNs(), keys, detail.appFilter()).stream()
+      .map(appData -> applicationFromCacheData(appData, detail))
+      .collect(toSet());
+  }
+
+  public Optional<CloudFoundryApplication> findApplicationByKey(String key, Detail detail) {
+    return Optional.ofNullable(cacheView.get(APPLICATIONS.getNs(), key, detail.appFilter()))
+      .map(appData -> applicationFromCacheData(appData, detail));
+  }
+
+  private CloudFoundryApplication applicationFromCacheData(CacheData appData, Detail detail) {
+    CloudFoundryApplication app = objectMapper.convertValue(appData.getAttributes().get("resource"), CloudFoundryApplication.class);
+    if (detail.equals(Detail.NONE))
```

Comment:
Can we do checkstyle or something?

Reply:
It wouldn't have to be complete. Just "parens for blocks" and "unused imports" or something like that to start. Stylistic preferences aren't always apparent to me otherwise.

Reply:
Good point, yes that's would be helpful